File: src/java/src/MultipleInheritance.java
//: src/java/src/MultipleInheritance.java
import java.util.List;

interface Base {
  void eat();
}

class Person2 extends Person implements Base {
  @Override public void eat() { consume(); }
}

class Robot2 extends Robot implements Base {
  @Override public void eat() { charge(); }
}

public class MultipleInheritance {
  static void nourish(Base subject) { subject.eat();  }
  public static void main(String... args) {
    List.of(new Person2(), new Robot2())
        .forEach(MultipleInheritance::nourish);
  }
}

File: src/kotlin/src/main/kotlin/MultipleInheritance.kt
//: src/kotlin/src/main/kotlin/MultipleInheritance.kt
// Can also do this in Java
import disjoint.*

interface Base {
    fun eat()
}

class Person2 : Base, Person() {
    override fun eat() = consume()
}

class Robot2 : Base, Robot() {
    override fun eat() = charge()
}

fun nourish(subject: Base) = subject.eat()

fun main() = listOf(Person2(), Robot2()).forEach { nourish(it) }
